rockin/plugins/conveyor_belt/conveyor_belt_thread.cpp

```diff
@@ -104,24 +123,42 @@ void ConveyorBeltThread::init()
 
 void ConveyorBeltThread::finalize()
 {
-    delete zmq_context_;
-    delete zmq_publisher_;
-    delete zmq_subscriber_;
+    if (mode_ == "real")
+    {
+        delete zmq_context_;
+        delete zmq_publisher_;
+        delete zmq_subscriber_;
 
-    zmq_context_ = NULL;
-    zmq_publisher_ = NULL;
-    zmq_subscriber_ = NULL;
+        zmq_context_ = NULL;
+        zmq_publisher_ = NULL;
+        zmq_subscriber_ = NULL;
+    }
 }
 
 void ConveyorBeltThread::loop()
 {
-    receiveAndBufferStatusMsg();
+    if (mode_ == "real")
+    {
+        receiveAndBufferStatusMsg();
+    } else if (mode_ == "simulation")
+    {
+        fawkes::MutexLocker lock(last_status_mutex_);
```

Comment:
Originally, I had all of the simulation inside of receiveAndBufferStatusMsg. 

But I moved it here because instead because it doesn't have anything to do with the functions name recieve and buffer status msgs.